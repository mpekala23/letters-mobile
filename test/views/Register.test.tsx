import React from "react";
import { RegisterScreen } from "@views";
process.env.MOCK_API_IP = "test_mock_api";
import { register } from "@api";
import {
  render,
  fireEvent,
  toJSON,
  getByPlaceholderText,
} from "@testing-library/react-native";
import fetchMock from "jest-fetch-mock";

jest.mock("@api", () => ({
  register: jest.fn(),
}));

const setup = (response = {}) => {
  const navigation = { navigate: jest.fn() };
  if (Object.keys(response).length > 0) {
    fetchMock.mockOnce(JSON.stringify(response));
  }
  return render(<RegisterScreen navigation={navigation} />);
};

describe("Register screen", () => {
  it("should match snapshot", () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it("should have register button be disabled until all fields are valid", async () => {
    const { getByPlaceholderText, getByText } = setup();
    fireEvent.press(getByText("Register"));
    expect(register).toHaveBeenCalledTimes(0);
    fireEvent.changeText(getByPlaceholderText("First Name"), "Team");
    fireEvent.changeText(getByPlaceholderText("Last Name"), "Ameelio");
    fireEvent.changeText(
      getByPlaceholderText("Cell Phone Number"),
      "4324324432"
    );
    fireEvent.changeText(getByPlaceholderText("Address Line 1"), "Address");
    fireEvent.changeText(getByPlaceholderText("Country"), "USA");
    fireEvent.changeText(getByPlaceholderText("Zip Code"), "12345");
    fireEvent.changeText(getByPlaceholderText("City"), "New Haven");
    fireEvent.changeText(getByPlaceholderText("State"), "CT");
    fireEvent.changeText(getByPlaceholderText("E-mail Address"), "team@ameelio.org");
    fireEvent.changeText(getByPlaceholderText("Password"), "GoodPassword1");
    fireEvent.press(getByText("Register"));
    expect(register).toHaveBeenCalledTimes(1);
  });
});
