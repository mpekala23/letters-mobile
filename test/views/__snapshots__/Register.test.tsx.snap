// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <ScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingBottom": 100,
          "paddingTop": 50,
        }
      }
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "backgroundColor": "#3490dc",
              "borderRadius": 50,
              "height": 100,
              "overflow": "hidden",
              "width": 100,
            }
          }
          testID="clickable"
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontStyle": "italic",
                "fontWeight": "400",
              },
              Object {
                "marginTop": 5,
              },
            ]
          }
        >
          Click to upload profile image.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D0E5FD",
            "borderColor": "#BEDAFC",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 15,
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontWeight": "600",
              },
              Object {
                "color": "#124181",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          We value your privacy. We will only use your address information as the return address for the mailed letter.
        </Text>
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "marginBottom": 10,
          }
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
              },
              Object {
                "color": "#3490dc",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Already have an account?
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.7}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3490dc",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {},
            Object {},
            Object {},
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
              Object {},
            ]
          }
        >
          Dev Skip
        </Text>
      </TouchableOpacity>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="First Name"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Last Name"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "width": "100%",
            }
          }
          testID="parent"
        >
          <ScrollView
            keyboardShouldPersistTaps="always"
            scrollEnabled={false}
            style={
              Array [
                Object {
                  "flex": 1,
                  "width": "100%",
                },
                Object {},
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="Cell Phone Number"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "height": 0,
                  "width": "100%",
                  "zIndex": 5,
                }
              }
            >
              <View />
            </View>
          </ScrollView>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Address Line 1"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Address Line 2"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Country"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Zip Code"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="City"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 60,
            "marginBottom": 0,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="State"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 200,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="E-mail Address"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 10,
            "width": "100%",
          }
        }
        testID="parent"
      >
        <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={false}
          style={
            Array [
              Object {
                "flex": 1,
                "width": "100%",
              },
              Object {},
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Password"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "borderColor": "#282828",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "color": "#282828",
                  "fontSize": 18,
                  "height": 50,
                  "padding": 10,
                  "width": "100%",
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </ScrollView>
      </View>
      <TouchableOpacity
        activeOpacity={1}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3490dc",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "grey",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
            Object {},
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
            ]
          }
        >
          Register
        </Text>
      </TouchableOpacity>
    </ScrollView>
  </View>
</View>
`;
